storage/series.go

```diff
@@ -217,3 +256,33 @@ type errChunksIterator struct {
 func (e errChunksIterator) At() chunks.Meta { return chunks.Meta{} }
 func (e errChunksIterator) Next() bool      { return false }
 func (e errChunksIterator) Err() error      { return e.err }
+
+// ExpandSamples iterates over all samples in the iterator, buffering all in slice.
+// Optionally it takes samples constructor, useful when you want to compare sample slices with different
+// sample implementations.
+func ExpandSamples(iter chunkenc.Iterator, newSampleFns ...func(t int64, v float64) tsdbutil.Sample) ([]tsdbutil.Sample, error) {
+	newSample := func(t int64, v float64) tsdbutil.Sample { return sample{t, v} }
+	if len(newSampleFns) > 0 {
+		newSample = newSampleFns[0]
+	}
+
+	var result []tsdbutil.Sample
+	for iter.Next() {
+		t, v := iter.At()
+		// NaNs can't be compared normally, so substitute for another value.
```

Comment:
This is unittest code that shouldn't be in a non-test file as it's dangerous to use generally.

Reply:
Well, there is no other way to share test helpers then this. 

What do you think about moving it to some `testutil` for series? We did similar on Thanos: https://github.com/thanos-io/thanos/tree/master/pkg/store/storepb/testutil (:

Reply:
I'm not against that.

Reply:
Still TODO in separate PR (: